Entities/MOSParticle.cpp

```diff
@@ -1,438 +1,204 @@
-//////////////////////////////////////////////////////////////////////////////////////////
-// File:            MOSParticle.cpp
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Source file for the MOSParticle class.
-// Project:         Retro Terrain Engine
-// Author(s):       Daniel Tabar
-//                  data@datarealms.com
-//                  http://www.datarealms.com
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Inclusions of header files
-
 #include "MOSParticle.h"
 #include "Atom.h"
-#include "RTEManagers.h"
-#include "RTETools.h"
+#include "PostProcessMan.h"
 
 namespace RTE {
 
-ConcreteClassInfo(MOSParticle, MovableObject, 200)
+	ConcreteClassInfo(MOSParticle, MovableObject, 200)
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void MOSParticle::Clear() {
+		m_Atom = 0;
+		m_SpriteAnimMode = OVERLIFETIME;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int MOSParticle::Create() {
+		if (MOSprite::Create() < 0) {
+			return -1;
+		}
+		if (!m_Atom) { m_Atom = new Atom(); }
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int MOSParticle::Create(const MOSParticle &reference) {
+		MOSprite::Create(reference);
+
+		m_Atom = new Atom(*(reference.m_Atom));
+		m_Atom->SetOwner(this);
+
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int MOSParticle::ReadProperty(std::string propName, Reader &reader) {
+		if (propName == "Atom") {
+			if (!m_Atom) { m_Atom = new Atom; }
+			reader >> *m_Atom;
+			m_Atom->SetOwner(this);
+		} else {
+			// See if the base class(es) can find a match instead
+			return MOSprite::ReadProperty(propName, reader);
+		}
+		return 0;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int MOSParticle::Save(Writer &writer) const {
+		MOSprite::Save(writer);
+
+		// TODO: Make proper save system that knows not to save redundant data!
+		/*
```

Comment:
Since entity writers are only relevant to editors I believe this is about constantly writing the whole preset every time it's used instead of a simple CopyOf. Not sure, don't really care at the moment.
And Atom is set in ini so it totally has to exist there.

Reply:
Oh right, MOSP has a sprite so it's not just one atom. Fair enough, would need that.
And okay, if it's about something more in depth then nevermind. I guess leave it be.